src/sandstorm/supervisor.capnp

```diff
@@ -266,6 +266,24 @@ struct ApiTokenOwner {
 
       title @7 :Text;
       # Title as chosen by the user.
+
+      appTitle @8 :Util.LocalizedText;
+      # A copy of the app name for the corresponding UIView for presentation in the grain list.
+
+      lastUsed @9 :Int64;
+      # The last time the user used this API token with the associated grain, in milliseconds
+      # since the epoch (equivalent to javascript's new Date().getTime())
+
+      icon :group {
```

Comment:
Seems like `lastUsed` and `title` are fields that owner.user.userId owns, whereas `appTitle`, `icon`, and `appId` are all denormalized grain information (which are owned by the grain owner).  I think I'd want to divide them on those lines.

And yes, appId should be in a union with icon.